src/pudl/transform/eia860.py

```diff
@@ -426,6 +418,18 @@ def _core_eia860__generators_energy_storage(
     return storage_df
 
 
+def drop_null_generator_id(
+    df: pd.DataFrame, num_of_expected_nulls: int
+) -> pd.DataFrame:
+    """Drop a prescribed number of records with null generator IDs."""
+    # there is one record that has a null gen id. ensure there isn't more before dropping
+    if len(null_gens := df[df.generator_id.isnull()]) > num_of_expected_nulls:
+        raise AssertionError(
+            f"Expected {num_of_expected_nulls} or zero records with a null generator_id but found {null_gens}"
+        )
+    return df.dropna(subset=["generator_id"])
```

Comment:
this null dropping was about to be happening in a few places so i tried to move it here